wrappers/src/shared_realm_cs.cpp

```diff
@@ -128,7 +128,13 @@ Realm::Config get_shared_realm_config(Configuration configuration, SyncConfigura
 
 inline SharedRealm* new_realm(SharedRealm realm)
 {
-    if (!realm->refresh()) {
+    if (!realm->config().immutable()) {
```

Comment:
This if statement can be reduced to:

```cpp
// If a Realm is immutable, it can't be refreshed
// If we can't refresh the Realm, make sure we begin a read transaction
// as a lot of functionality expects an active read transaction and OS
// doesn't always start one automatically.
if (realm->config().immutable() || !realm->refresh()) {
    realm->read_group();
}
```